jib-core/src/main/java/com/google/cloud/tools/jib/http/FailoverHttpClient.java

```diff
@@ -312,6 +333,23 @@ private Response call(String httpMethod, URL url, Request request, HttpTransport
         httpTransport
             .createRequestFactory()
             .buildRequest(httpMethod, new GenericUrl(url), request.getHttpContent())
+            .setIOExceptionHandler(
+                new HttpBackOffIOExceptionHandler(new ExponentialBackOff()) {
+                  @Override
+                  public boolean handleIOException(HttpRequest request, boolean supportsRetry)
+                      throws IOException {
+                    boolean result =
+                        enableRetries && super.handleIOException(request, supportsRetry);
+                    String requestUrl = request.getRequestMethod() + " " + request.getUrl();
+                    if (result) { // google-http-client does not log that properly so let's
+                                  // compensate it
+                      logger.accept(LogEvent.warn(requestUrl + " failed and will be retried"));
+                    } else {
+                      logger.accept(LogEvent.warn(requestUrl + " failed and will NOT be retried"));
```

Comment:
LogEvent.debug

Reply:
Can we keep info? debug means it will never be seen whereas it must be seen to report the server has issues. Happy to work on a config to change the level later if it bother users but strongly think the default should be visible without any log tuning.

Reply:
The user will see the IOException anyway, right? (BTW, as I mentioned in my earlier comment, I'm only taking about the `else` branch.) I think that is enough and seeing "will NOT be retried" every time on a failure, particularly if `supportsRetry` is false seems redundant. And generally speaking, we tend to be a bit sparing for WARN if it is not something actionable.

However, if this is never logged when for a no-error case (i.e., everything is working successfully), I can accept that. Can you confirm?

Reply:
Yes, it is only logged when there was an error and both cases are important (why wasnt it retried and why is it slow or replay basically).